systemtests/src/main/java/com/github/streamshub/systemtests/setup/StrimziOperatorSetup.java

```diff
@@ -226,66 +216,27 @@ private ClusterRoleBinding[] getBundleClusterRoleBindings() {
             .toArray(ClusterRoleBinding[]::new);
     }
 
-    private List<File> fetchStrimziYamlFiles() {
-        List<File> yamlFiles = new ArrayList<>();
-        for (String fileUrl : extractYamlFileLinksFromGithubDir()) {
-            String fileName = fileUrl.substring(fileUrl.lastIndexOf('/') + 1).replace(".yaml", "");
-            LOGGER.debug("Fetching Strimzi YAML file: {}", fileName);
-            Path tempFile = null;
+    private List<String> getStrimziYamlFilesContent() {
+        LOGGER.debug("Extracting Strimzi YAML file links from GitHub url: {}", resourceURL);
+        for (int attempt = 0; attempt < 10; attempt++) {
             try {
-                tempFile = Files.createTempFile(fileName + "-tmp_", ".yaml");
-                Files.write(tempFile, SetupUtils.getYamlFileContent(fileUrl).getBytes());
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
-            yamlFiles.add(tempFile.toFile());
-        }
-        return yamlFiles;
-    }
-
-    private List<String> extractYamlFileLinksFromGithubDir() {
-        LOGGER.debug("Extracting Strimzi YAML file links from GitHub url: {}", Environment.STRIMZI_OPERATOR_CRDS_URL);
-        int maxRetries = 10;
-        int attempt = 0;
-
-        while (attempt < maxRetries) {
-            try {
-                BufferedReader htmlWithLinksFromDir = new BufferedReader(
-                    new InputStreamReader(new URL(Environment.STRIMZI_OPERATOR_CRDS_URL).openStream()));
-
-                StringBuilder content = new StringBuilder();
-                String inputLine;
-
-                while ((inputLine = htmlWithLinksFromDir.readLine()) != null) {
-                    content.append(inputLine);
-                }
-
-                htmlWithLinksFromDir.close();
-
-                List<String> yamlFileLinks = new ArrayList<>();
-                Matcher matcher = Pattern.compile("href=\"(\\/[a-zA-Z0-9\\.\\/\\-]+\\/.{3}[a-zA-Z\\-]+\\d?\\.yaml)\"")
-                    .matcher(content.toString());
-
-                while (matcher.find()) {
-                    final String rawUrl = "https://raw.githubusercontent.com" + matcher.group(1)
-                        .replace("/blob/", "/");
-                    if (!yamlFileLinks.contains(rawUrl)) {
-                        yamlFileLinks.add(rawUrl);
-                    }
-                }
+                // Github row locator of current dir
+                List<String> yamlFileLinks = Jsoup.parse(new URL(resourceURL), 10_000)
+                    .select("tr.react-directory-row div.react-directory-filename-cell a.Link--primary")
```

Comment:
This seems like it could be brittle by relying on the GitHub UI. Isn't this fetching the same files that are concatenated and attached to the Strimzi releases?

Reply:
It's possible to fetch from the release yaml. My initial idea was to be able to work with all strimzi branches, but they are release branches anyway. So i'll change it to that one specific file. Thanks!